libstuff/libstuff.cpp

```diff
@@ -142,6 +157,58 @@ vector<string> SException::details() const noexcept {
     return stack;
 }
 
+void SSyslogSocketDirect(int priority, const char *format, ...) {
+    int socketError = 0;
+    static const size_t MAX_MESSAGE_SIZE = 8 * 1024;
+
+    // This block shouldn't be required, but it kicks in if this is used from somewhere that fails to call
+    // `SInitialize`. I figured this is better than just losing the log.
+    if (SSyslogSocketFD == 0) {
+        SSyslogSocketAddr.sun_family = AF_UNIX;
+        strcpy(SSyslogSocketAddr.sun_path, "/run/systemd/journal/syslog");
+        SSyslogSocketFD = -1;
```

Comment:
why set this to -1 here only to drop into the next if? why not just move L174 here, and remove the if from L172 to L175?

Reply:
It’s not better really, it just matches exactly what we do in SInitialize, which this is supposed to make up for not calling.